### ISSUE TITLE
[CRITICAL] Firebase FCM API has been shut down / Android push notifications not working

### ISSUE THREAD

**Issue by fruworg** [BelledonneCommunications/flexisip#196]
Hi, Firebase FCM API is [shutted down](https://firebase.google.com/docs/cloud-messaging/migrate-v1). Push notifications did not work.

We are using the latest stable version of flexisip (2.3.4).

Log:
```
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>]: sending request[<REDACTED>]:
HttpMessage : 
    HttpMessage header : 
:method = POST
:scheme = https
:path = /fcm/send
:authority = fcm.googleapis.com:443
content-type = application/json
authorization = key=<REDACTED>

    HttpMessage body : 
{
	"to":<REDACTED>",
	"time_to_live": 90,
	"priority":"high",
	"data":{
		"uuid":"<urn:uuid:<REDACTED>>",
		"from-uri":"",
		"display-name":"",
		"call-id":"<REDACTED>",
		"sip-from":"",
		"loc-key":"",
		"loc-args":"",
		"send-time":"2024-08-29 <REDACTED>"
		"custom-payload":{}
	}
}
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][0]: SETTINGS frame sent (6B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: HEADERS frame sent (170B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: DATA frame sent (488B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: request[<REDACTED>] submitted
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][0]: SETTINGS frame received (18B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][0]: server settings received
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][0]: WINDOW_UPDATE frame received (4B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][0]: SETTINGS frame sent (0B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][0]: SETTINGS frame received (0B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: HEADERS frame received (249B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: DATA frame received (101B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: DATA frame received (0B)
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: stream closed without error
2024-08-29 <REDACTED> flexisip-debug-Http2Client[<REDACTED>][1]: response received for HttpRequest[<REDACTED>]:
HttpMessage : 
    HttpMessage header : 
:status = 301
date = Thu, 29 Aug 2024 12:01:43 GMT
expires = Thu, 29 Aug 2024 12:01:43 GMT
cache-control = private, max-age=0
x-content-type-options = nosniff
x-frame-options = SAMEORIGIN
content-security-policy = frame-ancestors 'self'
x-xss-protection = 1; mode=block
content-length = 101
server = GSE
content-type = text/html; charset=UTF-8
alt-svc = h3=":443"; ma=<REDACTED>,h3-29=":443"; ma=<REDACTED>

    HttpMessage body : 
{"error":"Deprecated endpoint, see https:\/\/firebase.google.com\/docs\/cloud-messaging\/migrate-v1"}

```

**Comment by Failxxx** [comment]
Hello @fruworg,
Thank you for taking the time to report a problem with Flexisip.

We anticipated this problem months ago, here is how to proceed to use the new Firebase FCM API : [migration guide](https://wiki.linphone.org/xwiki/wiki/public/view/Flexisip/Configuration/Push%20notifications/Firebase%20legacy%20API%20to%20Firebase%20v1%20migration%20guide/).
It is available since Flexisip 2.3.3.

Sincerely,
Félix.

**Comment by fruworg** [comment]
> Hello @fruworg, Thank you for taking the time to report a problem with Flexisip.
> 
> We anticipated this problem months ago, here is how to proceed to use the new Firebase FCM API : [migration guide](https://wiki.linphone.org/xwiki/wiki/public/view/Flexisip/Configuration/Push%20notifications/Firebase%20legacy%20API%20to%20Firebase%20v1%20migration%20guide/). It is available since Flexisip 2.3.3.
> 
> Sincerely, Félix.

thank you!
